tools/clang/test/HLSLFileCheck/hlsl/operators/binary/shr.hlsl

```diff
@@ -0,0 +1,13 @@
+// RUN: %dxc -E main -T ps_6_0  %s | FileCheck %s
+
+// Make sure generate ashr and lshr
+// CHECK:ashr
+// CHECK:lshr
+int i;
+
+int half_btf(int w,int bit) { return (w + (1<<bit)) >> bit; }
+int half_btf2(int w,int bit) { return (w + (1U<<bit)) >> bit; }
```

Comment:
Shouldn't we have a case for literal and uint?
```
	int half_btf3(int w, uint bit) { return (w + (1<<bit)) >> bit; }
```

Plus we should test for the warning.  That requires something to force a compilation failure as well (like missing output semantic for instance), so we can check the error/warning output.  This would need to be a separate entry with separate RUN lines, or a separate test, of course.